copycat/api/src/main/java/org/apache/kafka/copycat/data/SchemaBuilder.java

```diff
@@ -346,7 +347,7 @@ public Schema valueSchema() {
      * @return the {@link Schema}
      */
     public Schema build() {
-        return new CopycatSchema(type, isOptional(), defaultValue, name, version, doc, fields, keySchema, valueSchema);
+        return new CopycatSchema(type, isOptional(), defaultValue, name, version, doc, fields == null ? null : Collections.unmodifiableList(fields), keySchema, valueSchema);
```

Comment:
Most schemas don't have fields -- all the primitive types, arrays, maps. i.e. everything but structs. In all those cases the list of fields is null (could just as easily be an empty list, especially since I think I have all code paths in Schema that could access it incorrectly throw exceptions if the schema type isn't a struct, but having an empty list doesn't benefit us at all).

This just got added because it's a guard against one of the few places where Schemas were still technically mutable. We could also move this into the CopycatSchema constructor instead, now that I think about it.